src/main/java/io/confluent/connect/hdfs/DataWriter.java

```diff
@@ -505,7 +505,10 @@ public Partitioner getPartitioner() {
 
   public Map<TopicPartition, Long> getCommittedOffsets() {
     for (TopicPartition tp : assignment) {
-      offsets.put(tp, topicPartitionWriters.get(tp).offset());
+      Long committedOffset = topicPartitionWriters.get(tp).committedOffset();
+      if (committedOffset != null) {
+        offsets.put(tp, committedOffset);
+      }
```

Comment:
We only include the offset for topic partitions with records that were committed by the TopicPartitionWriter instances to HDFS. If `committedOffset()` returns null, we don't even include it.

Despite the name, the previous incarnation of this method didn't return only the committed offsets; as mentioned in another comment, the offset of the first record each TopicPartitionWriter processed would be returned, whether or not this was actually written. And because this method was only used in tests, I felt it was acceptable to change the behavior to return only the committed offset.